Engine/DataFeeds/Enumerators/FillForwardEnumerator.cs

```diff
@@ -266,6 +266,14 @@ protected virtual bool RequiresFillForwardData(TimeSpan fillForwardResolution, B
             foreach (var item in GetSortedReferenceDateIntervals(previous, fillForwardResolution, _dataResolution))
             {
                 var potentialBarEndTime = RoundDown(item.ReferenceDateTime + item.Interval, item.Interval);
+                if (potentialBarEndTime == previous.EndTime)
+                {
+                    // time didn't advance, the interval we added is being swallowed by a daylight time change
+                    // so lets use interval * 2, and this should give us 1 rounded interval as result, which is what we want
+                    potentialBarEndTime = RoundDown(
+                        item.ReferenceDateTime + item.Interval + item.Interval,
```

Comment:
> I feel like this shouldn't fix it

Why? This only happens for `Resolution.Daily/Time.OneDay interval` since thats the resolution which will round down hours.
I've updated it to use the a `TimeZoneOffsetProvider` instance which also solved the issue, seems like a cleaner solution and did some performance tests and saw no significant overhead being added 👍

Reply:
you mention round down hours, but what about time zones that aren't perfect hours? for example, India is +10.5 hours from UTC.